src/porepy/numerics/ad/operators.py

```diff
@@ -1234,15 +1132,232 @@ def _parse_other(self, other):
+        iterate.
+
+        This attribute is only available in operators representing previous time steps.
+
+        """
+
+        self._time_step_index: int = -1
+        """Time step index, starting with -1 (current time) and increasing for previous
+        time steps."""
+
+    @property
+    def prev_time(self) -> bool:
+        """True, if the operator represents a previous time-step."""
+        return True if self._time_step_index > -1 else False
+
+    @property
+    def time_step_index(self) -> Optional[int]:
+        """Returns the time step index this instance represents.
+
+        - None indicates this is an operator at the current time step
+        - 0 represents the first previous time step
+        - 1 represents the next time step further back in time
+        - ...
+
+        Note:
+            The public time step index is ``None``, if the operator represents the
+            current time.
+
+        """
+        return self._time_step_index if self._time_step_index >= 0 else None
+
+    def previous_timestep(
+        self: _TimeDependentOperator, steps: int = 1
+    ) -> _TimeDependentOperator:
+        """Returns a copy of the time-dependent operator with an advanced time-step
+        index.
+
+        Time-dependent operators do not invoke the recursion (like the base class),
+        but represent a leaf in the recursion tree.
+
+        Note:
+            You cannot create operators at the previous time step from operators which
+            are at some previous iterate. Use the :attr:`original_operator` instead.
+
+        Parameters:
+            steps: ``default=1``
+
+                Number of steps backwards in time.
+
+        Raises:
+            ValueError: If this instance represents an operator at a previous iterate.
+            AssertionError: If ``steps`` is not strictly positive.
+
+        """
+        if isinstance(self, IterativeOperator):
+            if self.prev_iter:
+                raise ValueError(
+                    "Cannot create an operator representing a previous time step,"
+                    + " if it already represents a previous iterate."
+                )
+
+        assert steps > 0, "Number of steps backwards must be strictly positive."
+        # TODO copy or deepcopy? Is this enough for every operator class?
+        op = copy.copy(self)
+
+        # NOTE Use private time step index, because it is always an integer
+        # The public time step index is NONE for current time
+        # (which translates to -1 for the private index)
+        op._time_step_index = self._time_step_index + int(steps)
+
+        # keeping track to the very first one
+        if self.is_original_operator:
+            op.original_operator = self
+        else:
+            op.original_operator = self.original_operator
+
+        return op
+
+
+_TimeDependentOperator = TypeVar("_TimeDependentOperator", bound=TimeDependentOperator)
+
+
+class IterativeOperator(Operator):
```

Comment:
Analogous to the abstraction in `TimeDependentOperator`, but for iterative indices.

Reply:
Great!